plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/logging/RemoteQueryModifier.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.logging;
+
+import com.google.inject.Inject;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorSession;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_NON_TRANSIENT_ERROR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class RemoteQueryModifier
+{
+    public static final RemoteQueryModifier NONE = new RemoteQueryModifier(new RemoteQueryModifierConfig().setFormat("NOT_USED"))
```

Comment:
It looks like it would be good to have an interface `RemoteQueryModifier` and implementation `FormatBasedRemoteQueryModifier`.

Reply:
> setFormat("NOT_USED"))

This looks like a hack. Having an interface with a single method is not a big deal.

Reply:
> setFormat(“NOT_USED”)

Does it append anything ?